### PR TITLE
[RNMobile] Upgrade compile and target sdk version to Android API 33

### PR DESCRIPTION
**Pull request by fluiddot**
Related to https://github.com/wordpress-mobile/gutenberg-mobile/issues/5556.

**Related PRs:**
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/5789

## What?

Upgrade `react-native-aztec`, `react-native-bridge`, and `react-native-editor` projects to Android 13 (API 33). As well as the different React Native dependencies:
- AztecEditor-Android
- Libraries published via [`react-native-libraries-publisher`](https://github.com/wordpress-mobile/react-native-libraries-publisher/blob/2746ac55f78dd540efb2fa8089f1e05a2f7899a0/package.json#L3-L12)
- [`react-native-gesture-handler`](https://github.com/wordpress-mobile/react-native-gesture-handler)
- [`react-native-linear-gradient`](https://github.com/wordpress-mobile/react-native-linear-gradient)
- [`react-native-prompt-android`](https://github.com/wordpress-mobile/react-native-prompt-android)
- [`react-native-reanimated`](https://github.com/wordpress-mobile/react-native-reanimated)
- [`react-native-slider`](https://github.com/wordpress-mobile/react-native-slider)
- [`react-native-video`](https://github.com/wordpress-mobile/react-native-video)

## Why?

[Google Play requires apps to update to Android 13 (API level 33) before August 31, 2023](https://support.google.com/googleplay/android-developer/answer/11926878?hl=en).

## How?

* Bump `compileSdkVersion` and `targetSdkVersion`.
* Bump `buildToolsVersion` to match the React Native version ([reference](https://github.com/facebook/react-native/pull/35196/files)).
* Bump `kotlinVersion` to match the WPAndroid version ([reference](https://github.com/wordpress-mobile/WordPress-Android/blob/c5def47d3492c1266a61a8b2565310efe4d7f74f/settings.gradle#L2)).
* Bump `AztecEditor-Android` version to the one that upgraded to Android 13 ([reference](https://github.com/wordpress-mobile/AztecEditor-Android/releases/tag/v1.6.4)).
* Bump React Native dependencies
    * [Libraries published via `react-native-libraries-publisher`](https://github.com/wordpress-mobile/react-native-libraries-publisher/pull/17)
    * [`react-native-gesture-handler`](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/25)
    * [`react-native-linear-gradient`](https://github.com/wordpress-mobile/react-native-linear-gradient/pull/8)
    * [`react-native-prompt-android`](https://github.com/wordpress-mobile/react-native-prompt-android/pull/10)
    * [`react-native-reanimated`](https://github.com/wordpress-mobile/react-native-reanimated/pull/24)
    * [`react-native-slider`](https://github.com/wordpress-mobile/react-native-slider/pull/13)
    * [`react-native-video`](https://github.com/wordpress-mobile/react-native-video/pull/22)
    * [`react-native-hsv-color-picker`](https://github.com/wordpress-mobile/react-native-hsv-color-picker/pull/9) (this library depends on `react-native-linear-gradient`, so we need to update it too)
    * [Bump `react-native-webview` to version 11.26.1](https://github.com/wordpress-mobile/react-native-libraries-publisher/pull/20#top) (https://github.com/WordPress/gutenberg/pull/50731/commits/c45a12b3b660d5e98dc2f4b6dbf93e38e63cd1bb)
* Update RN native modules mocks to solve an issue with `react-native-webview` (https://github.com/WordPress/gutenberg/pull/50731/commits/aff6a533becf2d5027bfa58771c1d914b9aba748).

## Testing Instructions

This change doesn't affect a specific area, hence to test it we should basically smoke test the editor and check that works as expected.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A

### CODE REVIEW

File: packages/react-native-bridge/android/react-native-bridge/build.gradle [hunk 1]
```diff
@@ -69,13 +69,13 @@ dependencies {
...
-    implementation "com.github.wordpress-mobile:react-native-reanimated:${extractPackageVersion(packageJson, 'react-native-reanimated', 'dependencies')}"
-    implementation "com.github.wordpress-mobile:react-native-prompt-android:${extractPackageVersion(packageJson, 'react-native-prompt-android', 'dependencies')}"
+    implementation "com.github.wordpress-mobile:react-native-video:update~android-13-SNAPSHOT"
+    implementation "com.github.wordpress-mobile:react-native-linear-gradient:update~android-13-SNAPSHOT"
+    implementation "com.github.wordpress-mobile:react-native-slider:update~android-13-SNAPSHOT"
+    implementation "com.github.wordpress-mobile:react-native-reanimated:update~android-13-SNAPSHOT"
+    implementation "com.github.wordpress-mobile:react-native-prompt-android:update~android-13-SNAPSHOT"
 
-    implementation("com.github.wordpress-mobile:react-native-gesture-handler:${extractPackageVersion(packageJson, 'react-native-gesture-handler', 'dependencies')}", {
+    implementation("com.github.wordpress-mobile:react-native-gesture-handler:update~android-13-SNAPSHOT", {
```
Comment (fluiddot) [inline packages/react-native-bridge/android/react-native-bridge/build.gradle]: These changes will be reverted once the PRs in the forked repositories are merged.

File: test/native/setup.js [hunk 2]
```diff
@@ -19,24 +19,37 @@ global.ReanimatedDataMock = {
...
-			BEGAN: 'BEGAN',
-			FAILED: 'FAILED',
-			ACTIVE: 'ACTIVE',
-			END: 'END',
-		},
-		attachGestureHandler: jest.fn(),
-		createGestureHandler: jest.fn(),
-		dropGestureHandler: jest.fn(),
-		updateGestureHandler: jest.fn(),
+jest.mock( 'react-native', () => {
```
Comment (fluiddot) [inline test/native/setup.js:22]: I noticed when mocking the WebView native module that in some tests the mock was restored, leading to failures.
Mocking the native modules this way should prevent this issue.

File: packages/react-native-editor/ios/Podfile.lock
Comment (fluiddot) [inline packages/react-native-editor/ios/Podfile.lock:1]: I noticed that the version of the `react-native-slider` library hasn't been updated in this file. I checked the changes in the forked repository and realized that I overlooked updating one file. I've opened [this PR](https://github.com/wordpress-mobile/react-native-slider/pull/14) to fix it. @geriux let me know if you could take a look, thanks 🙇 !

Comment (fluiddot) [reply packages/react-native-editor/ios/Podfile.lock:1]: Solved in https://github.com/WordPress/gutenberg/pull/50731/commits/76436f56c20480bdb6db7413d085cc75a8601b5f.

**Review by geriux — APPROVED** [review]
LGTM! CI checks are passing and manual testing was done using the main apps, nice work! 🚀

**Comment by geriux** [comment]
What do you think about adding an internal note of the changes?

**Comment by fluiddot** [comment]
> What do you think about adding an internal note of the changes?

@geriux Sure, I'll add an internal note about this in `changelog` before merging. Thanks for reviewing the PR 🙇 !

**Pull request merged**